templates/terraform/bootstrap/secrets/main.tf

```diff
@@ -47,3 +47,11 @@ module "rds_master_secret_prod" {
   random_length = 32
   tags = map("rds", "${local.project}-prod")
 }
+
+module "sendgrid_api_key" {
+  source  = "../../modules/secret"
+  name = "${local.project}-sendgrid-<% index .Params `randomSeed` %>"
```

Comment:
This assumes the same api key will be shared for both staging / production, which I think is probably fine for now.